### PR TITLE
Feature/70 import range

### PR DESCRIPTION
**Pull request by StokesMIDE**
This reworks the `util.extractTime()` logic introduced in PR #60 to be more general, adding the ability to import a specific time range from a file.

### CODE REVIEW

File: idelib/importer.py [hunk 1]
```diff
@@ -354,14 +370,126 @@ def openFile(stream, updater=nullUpdater, parserTypes=elementParserTypes,
...
+                chId = blockStart = blockEnd = None
+                for subEl in el:
+                    if subEl.name == "ChannelIDRef":
+                        chId = subEl.value
+                    elif subEl.name == "StartTimeCodeAbs":
+                        blockStart = subEl.value
+                    elif subEl.name == "EndTimeCodeAbs":
+                        blockEnd = subEl.value
+
+                blockEnd = blockEnd or blockStart
```
Comment (CFlaniganMide) [inline idelib/importer.py:416]: Is the `EndTimeCodeAbs` value missing from single-sample channels, or is it just allowed to be missing?

**Review by CFlaniganMide — APPROVED** [review]
looks good, only one non-blocking question.

File: idelib/importer.py (hunk 1 above)
Comment (StokesMIDE) [reply idelib/importer.py:416]: It is allowed to be missing; this is what is done elsewhere in the Lab.

**Review by CFlaniganMide — APPROVED** [review]
new changes look good

**Pull request merged**